More 1.14 Progress

JRoy opened the pull request:
This pull request adds much of the mob and item work for you minus backward compatibility.

# Changelog:
* Targets 1.14-R0.1
* Adds 1.14 Mobs
* Adds 1.14 Items
  * Updated Signs as Well
* Adds 1.14 Blocks
* Fixes Misc Build Errors

# Notes While Testing:
* Signs work as intended
* Sometimes while moving after being AFK there is an error due to Spigot's new enforcement of async/sync events:
  * ```java.lang.IllegalStateException: AfkStatusChangeEvent cannot be triggered asynchronously from another thread.
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:502) ~[spigot-1.14.jar:git-Spigot-1eece4f-fe1199c]
        at com.earth2me.essentials.User.setAfk(User.java:463) ~[?:?]
        at com.earth2me.essentials.User.updateActivity(User.java:560) ~[?:?]
        at com.earth2me.essentials.EssentialsPlayerListener.delayedJoin(EssentialsPlayerListener.java:221) ~[?:?]
        at com.earth2me.essentials.EssentialsPlayerListener$1.run(EssentialsPlayerListener.java:198) ~[?:?]
        at org.bukkit.craftbukkit.v1_14_R1.scheduler.CraftTask.run(CraftTask.java:81) ~[spigot-1.14.jar:git-Spigot-1eece4f-fe1199c]
        at org.bukkit.craftbukkit.v1_14_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:54) [spigot-1.14.jar:git-Spigot-1eece4f-fe1199c]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:1.8.0_211]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:1.8.0_211]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_211]```

Hope this helps :)

JRoy committed:
Compile Against 1.14

JRoy committed:
Remove pre-release support from VersionUtil as it doesn't work

JRoy committed:
Fix 1.14 Build Errors

JRoy committed:
Update to 1.14 Mobs

JRoy committed:
Update to 1.14 Items

JRoy committed:
Update Commandkittycannon to support cats

JRoy committed:
Add 1.14 Blocks

mdcfe commented on Essentials/src/com/earth2me/essentials/MobData.java:
```diff
@@ -57,13 +58,19 @@
     GOLD_ARMOR_HORSE("goldarmor", EntityType.HORSE, EnumUtil.getMaterial("GOLDEN_HORSE_ARMOR", "GOLD_BARDING"), true),
     DIAMOND_ARMOR_HORSE("diamondarmor", EntityType.HORSE, EnumUtil.getMaterial("DIAMOND_HORSE_ARMOR", "DIAMOND_BARDING"), true),
     ARMOR_HORSE("armor", EntityType.HORSE, EnumUtil.getMaterial("IRON_HORSE_ARMOR", "IRON_BARDING"), true),
-    SIAMESE_CAT("siamese", EntityType.OCELOT, Ocelot.Type.SIAMESE_CAT, true),
```
This was left untouched as it breaks support for 1.13/1.12. I'll accept this for now but a better solution will be needed at some point.

mdcfe commented on Essentials/src/com/earth2me/essentials/utils/VersionUtil.java:
```diff
@@ -54,30 +52,26 @@ public static BukkitVersion fromString(String string) {
                 Integer.parseInt(patch),
-                Double.parseDouble(revision),
-                Integer.parseInt(prerelease));
+                Double.parseDouble(revision));
```
This will error on future prereleases; the ability to test on prereleases was the whole point of changing this.

mdcfe commented on Essentials/src/items.json:
```diff
@@ -9522,9 +9522,30 @@
     "spawnshulk": "shulker_spawn_egg",
-    "sign": {
-        "material": "SIGN"
+    "oak_sign": {
```
This file is automatically generated, so once the tool is updated these will be redundant.

JRoy replied:
Wish I knew that earlier :P

How do you generate that file anyway?

mdcfe reviewed (changes requested):
Thank for the PR, but the majority of the changes will need refactoring at some point:
* We're currently still aiming to support 1.12.2 (and 1.8 as long as feasible)
* We have an automated tool to update `items.json`
* We're not deliberately trying to break prereleases

JRoy committed:
Add proper prerelease support to VersionUtil

JRoy committed:
Add legacy support for ocelots for now

JRoy commented:
@md678685 I fixed pre-releases so now they actually work, and as for 1.12/1.13 support: I added back ocelots as the enum is just deprecated. But that obviously won't be good in the long run. I don't know how you want to maintain support so please do elaborate.

mdcfe commented on Essentials/src/com/earth2me/essentials/MobData.java:
```diff
@@ -107,7 +130,20 @@
     BETTY_TROPICAL_FISH("betty", EntityType.TROPICAL_FISH, TropicalFish.Pattern.BETTY, true),
-    CLAYFISH_TROPICAL_FISH("clayfish", EntityType.TROPICAL_FISH, TropicalFish.Pattern.CLAYFISH, true);
+    CLAYFISH_TROPICAL_FISH("clayfish", EntityType.TROPICAL_FISH, TropicalFish.Pattern.CLAYFISH, true),
+    BABY_FOX("baby", EntityType.FOX, Data.BABY, false),
```
Baby foxes should already be covered by `BABY_AGEABLE`.

JRoy committed:
Remove BABY_FOX from MobData

JRoy commented:
@md678685 Changes Applied 😄
